src/ResourceManager/Network/Commands.Network/NetworkSecurityGroup/SetAzureNetworkSecurityGroupCommand.cs

```diff
@@ -22,7 +22,7 @@
 
 namespace Microsoft.Azure.Commands.Network
 {
-    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetworkSecurityGroup"), OutputType(typeof(PSNetworkSecurityGroup))]
+    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetworkSecurityGroup", SupportsShouldProcess = true), OutputType(typeof(PSNetworkSecurityGroup))]
```

Comment:
@wdehrich To support this correctly, you will need to add a "ShouldProcess" if statement within your code that will not execute if users pass -WhatIf (it will simply show a message stating what would happen if the statement was executed.  See here for an example: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Aks/Commands.Aks/Commands/SetAzureRmAks.cs#L80.

Please also update the help files to get the two new parameters: https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md

Reply:
Please also update the changelog at src/Network/Commands.Network/Changelog.md